pkg/network/state.go

```diff
@@ -508,6 +522,26 @@ func (ns *networkState) GetStats() map[string]interface{} {
 	}
 }
 
+// Telemetry returns statistics that are included to evaluate internal behavior of system-probe
+func (ns *networkState) Telemetry(khits int64, kmisses int64, conntrackStats map[string]int64, dnsStats map[string]int64) map[string]int64 {
+	conntrackTotal := conntrackStats["registers_total"] + conntrackStats["registers_dropped"]
+	openedConns := ns.telemetry.connsOpened
+	tm := map[string]int64{
+		// TODO need to handle negative total (underflow) when counter rolls over?
+		"kprobes.triggered": khits - ns.telemetry.lastKprobesTriggered,
+		"kprobes.missed":    kmisses - ns.telemetry.lastKprobesMissed,
+		"conntrack.msgs":    conntrackTotal - ns.telemetry.lastConntrackTotal,
+		"dns.packets":       dnsStats["packets_processed"] - ns.telemetry.lastDnsPacketsProcessed,
+		"conns.opened":      openedConns - ns.telemetry.lastConnsOpened,
+	}
+	ns.telemetry.lastKprobesTriggered = khits
+	ns.telemetry.lastKprobesMissed = kmisses
+	ns.telemetry.lastConntrackTotal = conntrackTotal
+	ns.telemetry.lastDnsPacketsProcessed = dnsStats["packets_processed"]
+	ns.telemetry.lastConnsOpened = openedConns
+	return tm
+}
```

Comment:
Is there any reason this can't be a dedicated type instead of `map[string]int64` ? We might want to use floats in the future, and this type means callers need to know about the map keys.

Reply:
I can make that change. One benefit of the map is less churn on the protobuf definition, but that has its downsides as well.